pkgs/os-specific/linux/zsa-udev-rules/default.nix

```diff
@@ -20,6 +20,8 @@ stdenv.mkDerivation rec {
     mkdir -p $out/lib/udev/rules.d
     cp dist/linux64/50-oryx.rules $out/lib/udev/rules.d/
     cp dist/linux64/50-wally.rules $out/lib/udev/rules.d/
+    echo '# Voyager' >> $out/lib/udev/rules.d/50-wally.rules
+    echo 'SUBSYSTEMS=="usb", ATTRS{idVendor}=="3297", MODE:="0666", SYMLINK+="ignition_dfu"' >> $out/lib/udev/rules.d/50-wally.rules
```

Comment:
If you do `cat /etc/udev/rules.d/50-wally.rules` you'll see that the output is a bit different  from the documentation you are referring to. Specifically, `MODE:="0666"` becomes `TAG+="uaccess"`.
That is because the documentation doesn't incorporate the [latest](https://github.com/zsa/wally/commit/d9da6cd4a3f60dc4e6da20618019490632e15acb) changes to the rules. I think it is for security reasons that they do that, but I am by no means an expert. I also haven't tested those changes to my board. I don't really know how to do so, but I'm willing to do that if you point me in the right direction.

Reply:
I think I see what you mean. The other rules do have `TAG+="uaccess"` instead of `MODE...`. I can update this PR if that is the recommended approach. For testing, what I do is place update the package in my overlays array:

```nix
overlays = [
  (final: prev: {
    zsa-udev-rules = prev.zsa-udev-rules.overrideAttrs (old: {
      installPhase = old.installPhase + ''

        echo '# Keymapp Flashing rules for the Voyager' >> $out/lib/udev/rules.d/50-wally.rules
        echo 'SUBSYSTEMS=="usb", ATTRS{idVendor}=="3297", MODE:="0666", SYMLINK+="ignition_dfu"' >> $out/lib/udev/rules.d/50-wally.rules
      '';
    });
  })
];
```

Note that you need to restart your system for these udev changes to take effect.

Reply:
I have tested both with TAG and MODE but I still get the same error as before: `⠋ Press the reset button of your keyboard.2023/11/18 22:08:29 Error while extracting DFU Suffix: Invalid vendor or product id, expected 0x83:0x11 got 0x97:0x91`. But as @Shawn8901 mentioned the Voyager is not flashable anymore via Wally.

Reply:
The error code is saying that also. It's not complaining about permission or so, but says that the device in Bootmode has an unexpected product and vendor Id. https://github.com/zsa/wally-cli/blob/b0fafe52cc7fb9d55f2b968d4548c99917c7325c/usb.go#L16 that's where they are from in wally-cli.

Reply:
@gerpiet When I use `TAG+="uaccess"` training works but flashing does not not. When using `MODE...` both training and flashing work. Tested using google chrome.